src/interpreter_funcs/join_group.rs

```diff
@@ -0,0 +1,42 @@
+use std::sync::Arc;
+
+use crate::interpreter::{
+    Func, FuncError, FuncFlags, FuncInfo, ParamInfo, ParamRefinement, Ty, Value,
+};
+use crate::mesh::tools;
+
+pub struct FuncJoinGroup;
+
+impl Func for FuncJoinGroup {
+    fn info(&self) -> &FuncInfo {
+        &FuncInfo {
+            name: "Join Group",
+            return_value_name: "Joined Meshes",
+        }
+    }
+
+    fn flags(&self) -> FuncFlags {
+        FuncFlags::PURE
+    }
+
+    fn param_info(&self) -> &[ParamInfo] {
+        &[ParamInfo {
+            name: "Group",
+            refinement: ParamRefinement::MeshArray,
+            optional: false,
+        }]
+    }
+
+    fn return_ty(&self) -> Ty {
+        Ty::Mesh
+    }
+
+    fn call(&mut self, args: &[Value]) -> Result<Value, FuncError> {
+        let mesh_arc_array = args[0].unwrap_mesh_array();
+
+        // TODO: this doesn't work :(
+        let meshes = mesh_arc_array.iter();
```

Comment:
You need to `Deref` the `Arc` to a `&`. It's just one `.map()` away :wink:

https://doc.rust-lang.org/std/ops/trait.Deref.html 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use%20std%3A%3Aops%3A%3ADeref%3B%0Ause%20std%3A%3Async%3A%3AArc%3B%0A%0Astruct%20Thing%3B%0A%0Afn%20take%3C%27a%2C%20I%3E(things%3A%20I)%20where%20I%3A%20IntoIterator%3CItem%3D%26%27a%20Thing%3E%20%7B%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20things%20%3D%20vec!%5BArc%3A%3Anew(Thing)%2C%20Arc%3A%3Anew(Thing)%5D%3B%0A%20%20%20%20take(things.iter().map(%7Cthing%7C%20Arc%3A%3Aderef(%26thing)))%3B%0A%7D

Reply:
Hmm, now that I am thinking about it, it would be good to add that derefed iterator to `MeshArrayValue` directly... Maybe rename the current iterator to `iter_refcounted` and make the new `iter` do what you want here.